### DISCUSSION TITLE
Problem using SH1106 driver in Writer class

### DISCUSSION THREAD

**Discussion by jimwims** [micropython/micropython#11037] [Libraries & Drivers]
My board is an Adafruit QT Py ESP32 Pico running MicroPython v1.19.1 and using the Writer class in micropython-font-to-py as informed by micropython-nano-gui.   I want to use it with either an ssd1306 display or an sh1106 using the respective drivers on the nano-gui repository.  

All goes well with the ssd1306 driver.  But with the sh1106 driver, I get the following error:

`Traceback (most recent call last):
            File "main.py", line 18, in <module>
            File "drivers/sh1106/sh1106.py", line 269, in __init__
        AttributeError: 'int' object has no attribute 'init'
        MicroPython v1.19.1 on 2022-06-18; ESP32 module (spiram) with ESP32
`
Is there something wrong with the sh1106.py driver?  Or have I done something wrong?

**Comment by robert-hh** [comment]
Most probably you supply the wrong type of objects to the driver, like a pin number instead of a pin object. Can you show, how you use the sh1106 driver? It's where you instantiate the sh1106 object.

**Reply by jimwims** [reply]
Here is the code from the script up until that point.   One of the two drivers is selected in the third and forth lines from the bottom.  It works fine for the ssd11306. 

```py
from secrets import secrets
import utime
import urequests
from machine import Pin, I2C, RTC
import machine   #needed for machine.reset() 
import esp32
from ntptime import settime
from neopixel import NeoPixel

from writer_23 import Writer
import font10_23 as font10

from drivers.sh1106.sh1106 import SH1106_I2C as OLED       #uncomment for ssh1106
#from drivers.ssd1306.ssd1306 import SSD1306_I2C as OLED     #uncomment for ssd1306

i2c = I2C(1, scl=Pin(19), sda=Pin(22), freq=400000)
oled = OLED(128, 64, i2c, 0x3c)

```

**Reply by robert-hh** [reply]
The __init__() method of the SH1106_I2C class reads:
```
    def __init__(self, width, height, i2c, res=None, addr=0x3c,
                 rotate=0, external_vcc=False, delay=0):

```

So the 4th positional argument is taken as reset pin, not the address. Use:

`oled = OLED(128, 64, i2c, addr=0x3c)
`

**Reply by jimwims** [reply]
That solves the problem!  

Thank you very much for your help this time and for the many other contributions you have made to MicroPython.
